packages/core/src/types.ts

```diff
@@ -804,10 +799,7 @@ export interface MachineImplementationsSimplified<
 > {
   guards: Record<string, GuardPredicate<TContext, TEvent>>;
   actions: ActionFunctionMap<TContext, TEvent, TAction>;
-  actors: Record<
-    string,
-    ActorBehaviorCreator<TContext, TEvent> | AnyActorBehavior
-  >;
+  actors: Record<string, AnyActorBehavior>;
```

Comment:
As we might see in the diff here - now actors won't have access to `TContext` and `TEvent`. So it's not possible to "configure" their dynamic part (input) in the implementations object. That's only possible within the config itself, through the `invoke.input` property. However, at the moment, we are not able to provide typegen-based information there so this might get annoying to people. 

WDYT about allowing this to be:
```ts
// it's not a complete final type, just an illustration for this accepting `{ src, input }` 
actors: Record<string, AnyActorBehavior | { src: AnyActorBehavior, input: InputCreator<TContext, TEvent> }>; 
```

Reply:
The problem is that `input` might be different depending on where you come from:
```js
state1: {
  invoke: {
    src: 'something',
    input: (ctx) => /* get from context */
  }
}

state2: {
  invoke: {
    src: 'something',
    input: (_ctx, ev) => /* get from event */
  }
}
```

I don't understand the typegen information; `invoke.input` is only relevant to the invoked actor, not to the machine (which is the scope of typegen); can you clarify?

Reply:
The conceptual `TInput` always comes from a given actor type and is the same regardless of where and how it is instantiated. What might be different between instantiation sites, are the "arguments" used to resolve that same type. But that's exactly the same as with any other implementation type:
```ts
state1: {
  entry: 'something'
}

state2: {
  entry: 'something'
}
```
The final implementation of `something` just has to deal with all possible argument types. It's free to ignore `context` completely, and it's free to ignore `event` - but when it starts using the `event`, it just has to assume that it's a union of all possible event types that can lead to triggering this specific action.

Reply:
What should be the resolution for this?

Reply:
I propose this: https://github.com/statelyai/xstate/pull/3897